aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -532,15 +571,21 @@ module aptos_framework::fungible_asset {
         assert!(amount == 0, error::invalid_argument(EAMOUNT_IS_NOT_ZERO));
     }
 
-    fun deposit_internal<T: key>(store: Object<T>, fa: FungibleAsset) acquires FungibleStore {
+    fun deposit_internal<T: key>(store: Object<T>, fa: FungibleAsset) acquires FungibleStore, ConcurrentFungibleBalance {
         let FungibleAsset { metadata, amount } = fa;
         if (amount == 0) return;
 
         let store_metadata = store_metadata(store);
         assert!(metadata == store_metadata, error::invalid_argument(EFUNGIBLE_ASSET_AND_STORE_MISMATCH));
         let store_addr = object::object_address(&store);
-        let store = borrow_global_mut<FungibleStore>(store_addr);
-        store.balance = store.balance + amount;
+
+        if (concurrent_fungible_balance_exists(store_addr)) {
+            let balance = borrow_global_mut<ConcurrentFungibleBalance>(store_addr);
+            aggregator_v2::add(&mut balance.balance, amount);
+        } else {
+            let store = borrow_global_mut<FungibleStore>(store_addr);
+            store.balance = store.balance + amount;
```

Comment:
We don't care about overflows? Interesting, why?

Reply:
will later do sth like
https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/coin.move#L547
cc @junkil-park